provider/defaultingress/ingress.go

```diff
@@ -0,0 +1,453 @@
+		state.ClusterRoutesTlsSecretRef = types.StringValue(tls)
+	} else {
+		state.ClusterRoutesTlsSecretRef = types.StringNull()
+	}
+	state.LoadBalancerType = types.StringValue(string(ingress.LoadBalancerType()))
+
+	return nil
+}
+
+func (r *DefaultIngressResource) updateIngress(ctx context.Context, state, plan *DefaultIngress,
+	clusterId string, clusterCollection *cmv1.ClustersClient) error {
+
+	if state == nil {
+		state = &DefaultIngress{Cluster: plan.Cluster}
+	}
+	// In case default ingress was not part of state till now and we want to set
+	// it we need to bring it first as we need to set specific id
+	if common.IsStringAttributeUnknownOrEmpty(state.Id) {
+		err := r.populateDefaultIngress(ctx, state)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !reflect.DeepEqual(state, plan) {
+		err := validateDefaultIngress(ctx, plan)
+		if err != nil {
+			return err
+		}
+		if plan == nil {
+			plan = &DefaultIngress{}
+		}
+
+		ingressBuilder := getDefaultIngressBuilder(ctx, plan)
+
+		if !reflect.DeepEqual(state.ClusterRoutesHostname, plan.ClusterRoutesHostname) {
+			ingressBuilder.ClusterRoutesHostname(plan.ClusterRoutesHostname.ValueString())
+		}
+		if !reflect.DeepEqual(state.ClusterRoutesTlsSecretRef, plan.ClusterRoutesTlsSecretRef) {
+			ingressBuilder.ClusterRoutesTlsSecretRef(plan.ClusterRoutesTlsSecretRef.ValueString())
+		}
+
+		ingress, err := ingressBuilder.Build()
+		if err != nil {
+			return err
+		}
+
+		ingressResp, err := clusterCollection.Cluster(clusterId).Ingresses().Ingress(state.Id.ValueString()).Update().
+			Body(ingress).SendContext(ctx)
+
+		if err != nil {
+			return err
+		}
+		if err := r.populateState(ingressResp.Body(), plan); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getDefaultIngressBuilder(ctx context.Context, state *DefaultIngress) *cmv1.IngressBuilder {
+	ingressBuilder := cmv1.NewIngress()
+	routeSelectors, err := common.OptionalMap(ctx, state.RouteSelectors)
+	if err != nil {
+		return nil
+	}
+	if routeSelectors == nil {
+		routeSelectors = map[string]string{}
+	}
+	ingressBuilder.RouteSelectors(routeSelectors)
+
+	excludedNamespace := common.OptionalList(state.ExcludedNamespaces)
+	ingressBuilder.ExcludedNamespaces(excludedNamespace...)
+
+	// wildcard policy can't be empty
+	if !common.IsStringAttributeUnknownOrEmpty(state.WildcardPolicy) {
+		ingressBuilder.RouteWildcardPolicy(cmv1.WildcardPolicy(state.WildcardPolicy.ValueString()))
+	}
+	// NamespaceOwnershipPolicy can't be empty
+	if !common.IsStringAttributeUnknownOrEmpty(state.NamespaceOwnershipPolicy) {
+		ingressBuilder.RouteNamespaceOwnershipPolicy(cmv1.NamespaceOwnershipPolicy(state.NamespaceOwnershipPolicy.ValueString()))
+	}
+	// LoadBalancer type can't be empty
+	if !common.IsStringAttributeUnknownOrEmpty(state.LoadBalancerType) {
+		ingressBuilder.LoadBalancerType(cmv1.LoadBalancerFlavor(state.LoadBalancerType.ValueString()))
+	}
+	return ingressBuilder
+}
+
+func ingressResource() map[string]schema.Attribute {
```

Comment:
Please merge this function into the Schema function, to keep the same convention as in other resources